### ISSUE TITLE
[Request] Provide media information for subtitle addons

### ISSUE THREAD

**Issue by ppaladin01** [elgatito/plugin.video.elementum#196]
## Expected Behavior

On Kodi Krypton, when clicking the Elementum subtitle downloader the tv show episode is recognized and the related subtitle available for download. Exemple: The downloader shows automatically TVShowXYZ S02E10 on the search box
## Current Behavior

On Kodi Leia, no subtitle is found and on the search box instead of the episode of the TV show being displayed, it shows the word "links" and of course no subtitle can be displayed.
## Steps to Reproduce (for bugs)

1.Navigate to Elementum TV Shows
2.Choose any tv show
3.When the stream begins click on download subtitle and select Elementum or any other subtitle addon
4.No subtitle is found

## Context

It works fine with movies, but since my device updated to Android Oreo, Elementum is only working properly on Kodi Leia.
## Your Environment

* Version used:

* Environment name and version:

* Operating System and version:

* Link to Kodi debug log:

Log:
https://pastebin.com/Aku1UvaB

**Comment by elgatito** [comment]
I can't see the problem. I try on stable Kodi and it uses proper query for subtitles search.

However, have to mention, Kodi allows setting OriginalTitle for movies, which is used for subtitles search. But for episodes it allows only localized Show name, which is later used in the query.

In your log subtitles use this query: `Query:2x10 The Passenger`

**Comment by ppaladin01** [comment]
Yes, that's the problem. On Kodi 17 when you browse a season and select and play an episode it's displayed like this: "TVShowName S02E10". So the subtitles services can find the proper file.

On Kodi 18 the same episode is displayed like this: "2x10 Episode Title". And of course the subtitles services won't find anything.

I used Elementum for months on Kodi 17 and SPMC on Android 6.0 without any issues, but unfortunately my device auto updated to Oreo and when I try to use Elementum on both versions, even 720p files get really choppy and slugish playback.

With Kodi 18 these issues are less visible, so I hoped I could stick with it.

But I assume there is not much what to do since it's not a stable version. Thanks for your good work and this awesome addon.

**Comment by elgatito** [comment]
@ppaladin01 can you make a log file with previous version to see how it worked for you?

**Comment by elgatito** [comment]
Interesting. On older Kodi:

```
SubtitlesIndex   [0mSubtitles payload: [{Query:Westworld S02E10 Hash: Size:0 IMDBId: Languages:eng}]
```

On newer Kodi:
```
SubtitlesIndex   [0mSubtitles payload: [{Query:2x10 The Passenger  Hash: Size:0 IMDBId: Languages:eng}]
```

**Comment by elgatito** [comment]
Somehow on new Kodi the item was resolved as Movie playback, not Episode:

```
02:07:10.771 T:18446744072633305456  NOTICE: [plugin.video.elementum] [36mDEBU  main         ▶ Notification     [0mGot notification from xbmc/Player.OnPlay: {"item":{"title":"2x10 The Passenger","type":"movie"},"player":{"playerid":1,"speed":1}}
```

**Comment by ppaladin01** [comment]
@elgatito your finding also explains why the episodes navigation menus are kinda broken.

**Comment by elgatito** [comment]
@ppaladin01 it's strange, but still not knowing the root of this situation. 
It can also be the bug in Kodi18, as it's in alpha still.

**Comment by elgatito** [comment]
@ppaladin01 you have wrong listings in the plugin when you browse episodes of a season/show?

**Comment by antonsoroko** [comment]
Same issue with Kodi 18 rc-1. This is very sad.

**Comment by elgatito** [comment]
Please, try version 0.0.66.

**Comment by antonsoroko** [comment]
@elgatito 

Elementum subtitles provider now works.
Others (addicted, opensubtitles, subscene) - don't :-(

Word "links" still is in the title (not sure if it is wrong or not).

Kodi 18 rc-2

**Comment by elgatito** [comment]
I'd appreciate a log file and screenshot. 
External subtitles' addons are getting title of a played video with their own logic.

**Comment by antonsoroko** [comment]
https://paste.ubuntu.com/p/BkZVnJNMtP/

![img_20181205_000608](https://user-images.githubusercontent.com/17964763/49473019-f6060900-f821-11e8-9518-f443eb0f04ab.jpg)

**Comment by antonsoroko** [comment]
только вот если нажать скачать - то пишет "open ..." (путь до конца не виден) ти потом что не может загрузить файл
https://paste.ubuntu.com/p/s9rBmDsk8T/

кажется такое же тут https://github.com/elgatito/plugin.video.elementum/issues/282

**Comment by elgatito** [comment]
```

0:05:21.772 T:18446744072725645680   DEBUG: service.subtitles.rvm.addic7ed: Using filename: links
00:05:21.773 T:18446744072725645680   DEBUG: service.subtitles.rvm.addic7ed: Filename links failed. Trying ListItem.Label...
00:05:21.774 T:18446744072725645680   DEBUG: service.subtitles.rvm.addic7ed: Using filename: Нечеловеческая природа
00:05:21.774 T:18446744072725645680   ERROR: service.subtitles.rvm.addic7ed: Unable to determine episode data for Нечеловеческая природа

```

```
00:05:33.282 T:18446744072725645680   DEBUG: ### [__main__] 0.246036 - Search_subscene='{'mansearch': False, 'episode': '7', 'temp': True, 'title': ' ', 'season': '14', 'tvshow': '', 'rar': False, 'year': '2018', 'file_original_path': u'http://127.0.0.1:65220/files/Supernatural.S14E07.HDTV.x264-SVA[eztv].mkv', '3let_language': ['eng', 'rus']}', filename='Supernatural.S14E07.HDTV.x264-SVA[eztv]', addon_version=1.7.2
00:05:33.284 T:18446744072725645680   DEBUG: ### [__main__] 0.248629 - Search movie =
00:05:33.285 T:18446744072725645680   DEBUG: ### [SubsceneUtilities] 0.249137 - Getting url: https://subscene.com/subtitles/title?q=&r=true
```

Every plugin resolves in a different way. One takes episode name - which is in Kodi language, not original language. One is trying to find according to the filename.

**Comment by antonsoroko** [comment]
@elgatito I think you could close this issue because it seems to be fixed.
there is a small bug https://github.com/elgatito/plugin.video.elementum/issues/318 with lower priority

Thank you!

**Comment by martinstz** [comment]
"Links" text showing in subtitles tab only when i use Elementum on Kodi 18.x. On other addons stay the release of movie. Can you fix it? Thank you in advance!

Kodi 17.x:
![1](https://user-images.githubusercontent.com/8360252/51612918-105b8980-1f2b-11e9-8324-bef9e77a8b83.png)

Kodi 18.x:
![2](https://user-images.githubusercontent.com/8360252/51612928-16ea0100-1f2b-11e9-966f-1a015d5dca05.png)

**Comment by elgatito** [comment]
@martinstz As you can read in comments, I have tried to change the way how streams are handled in Kodi, but Kodi changed somehow in Kodi 18 and old methods crash Kodi. 
I will not make any more efforts on this issue. 

Any PR to fix this issue are welcome (they should apply to Golang part, not python).

**Comment by elgatito** [comment]
This should be fixed with 0.1.25. I have checked with Opensubtitles addon and build-in Opensubtitles client.
All the information is in ListItems. Url is correct. 
I don't see what can be changed.
